exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserServer.java

```diff
@@ -255,11 +257,12 @@ void disableReadTimeout() {
       getChannel().pipeline().remove(BasicServer.TIMEOUT_HANDLER);
     }
 
-    void setUser(final UserToBitHandshake inbound) throws IOException {
+    void setUser(final UserToBitHandshake inbound, Map<String, String> sessionParams) throws IOException {
```

Comment:
Lets say there is an external custom authenticator.... It validates the credentials with that external system and loads contextual information from that external system. Examples:
  - Tenant_Id of the currently logged in user
  - Roles of the user logged in
  - Any user preference details that are loaded in the context of that external system
  - Etc..

User Properties, as I understood seem to be the configuration variables (E.g., drill-override.conf). These are known expected properties. Whereas Session Parameters may not be known properties.